storage/client.go

```diff
@@ -398,6 +398,15 @@ func (c Client) exec(verb, url string, headers map[string]string, body io.Reader
 		return nil, errors.New("azure/storage: error creating request: " + err.Error())
 	}
 
+	// if a body was provided ensure that the content length was set.
```

Comment:
This only works for io.LimitReader (plus the other reader types handled by `http.NewRequest()`). I suggest parsing the value from `headers["Content-Length"]` and use it (since it's passed down from `PutBlockWithLength`). This will make it more flexible and cover any type of io.Reader.